SparkPost/2web2ui
# TR-1533 Fix content editing modes

Renders editor in read only mode when viewing a published template; editable when viewing a draft. 

### What Changed
 - Render editors based on template status. Draft is editable, locked in published version
 - Add draft/published indicate at top bar
 - Fields in `Template Settings` tab are disabled when viewing published template
   - Delete button should be active
 - To distinguish (and deep link), modified route to add version
 - List and create pages are updated to take to new routes
 - Updated List page to use `list_name` to preserve original name in order to show correct messaging after delete. 

### How To Test
From Campaigns -> Templates
 - Open a draft template. You should be able to edit and top indicator should show draft
- Open a published template. Editing areas should be read only and indicated it's a published template like the image below. Template Settings should also be in read only mode
<img width="245" alt="Screen Shot 2019-06-18 at 6 51 29 PM" src="https://user-images.githubusercontent.com/14877079/59725029-1adc2f80-91fa-11e9-9a8b-bb3a2f03fa6f.png">
-  Create a template. After template creation, it should take you to `/edit/<id>/draft/content`
- Modify a draft (w/o publish) and see the two versions are rendered correctly
